generator/generator.go

```diff
@@ -82,6 +105,88 @@ func writeGoCode(w io.Writer, goCode *ygen.GeneratedGoCode) error {
 	return nil
 }
 
+// writeIfNotEmpty writes the string s to b if it has a non-zero length.
+func writeIfNotEmpty(b *bytes.Buffer, s string) {
+	if len(s) != 0 {
+		b.WriteString(s)
+	}
+}
+
+// codeOut describes an output file for Go code.
+type codeOut struct {
+	// contents is the code that is contained in the output file.
+	contents string
+	// oneoffHeader indicates whether the one-off header should be included in this
+	// file.
+	oneoffHeader bool
+}
+
+// makeOutputSpec generates a map, keyed by filename, to a codeOut struct containing
+// the code to be output to that filename. It allows division of a ygen.GeneratedGoCode
+// struct into a set of source files. It divides the methods, interfaces, and enumeration
+// code snippets into their own files. Structs are output into files dependent on the
```

Comment:
Is this a good split? (Based on the first name). How does the distribution look like, do we have a good spread?

Reply:
`N` continues to be the largest for OpenConfig, since it contains all the `NetworkInstance...` code. An alternate splitting strategy would be to do it at the next-level down from `NetworkInstance`, but `NetworkInstance_Protocol` would still end up being quite big -- and this would be quite a high number of files. Here's the breakdown of the `exampleoc` package (which contains almost the whole schema, and has`-generate_getters`, `-generate_rename` and `-generate_append`. It does not have `-exclude_state`.

```
[8:10] openconfig/ygot/splitoc   file-split ✔    9h46m ◒  
▶ du -sch pkg/oc/*
480K	pkg/oc/enum.go
160K	pkg/oc/enum_map.go
2.1M	pkg/oc/methods.go
2.5M	pkg/oc/schema.go
 24K	pkg/oc/structs_A.go
152K	pkg/oc/structs_B.go
 48K	pkg/oc/structs_C.go
4.0K	pkg/oc/structs_D.go
 80K	pkg/oc/structs_I.go
 24K	pkg/oc/structs_L.go
 88K	pkg/oc/structs_M.go
828K	pkg/oc/structs_N.go
 28K	pkg/oc/structs_O.go
 20K	pkg/oc/structs_R.go
 76K	pkg/oc/structs_S.go
 20K	pkg/oc/structs_T.go
276K	pkg/oc/union.go
6.8M	total
```